Build Error

While Running facing below error.
Flutter verion: 3.7.5
Visual Studio: 2022


```C:/Users/Administrator/Documents/codecommit/url_host/build/windows/plugins/webview_windows/Debug/webview_windows_plugin.exp
[ +685 ms] std.ixx.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) void (__cdecl* std::_Raise_handler)(class
stdext::exception const &)" (__imp_?_Raise_handler@std@@3P6AXAEBVexception@stdext@@@ZEA::<!std>) referenced in function "public: void    
__cdecl stdext::exception::_Raise(void)const " (?_Raise@exception@stdext@@QEBAXXZ::<!std>)
[C:\Users\Administrator\Documents\codecommit\url_host\build\windows\plugins\webview_windows\webview_windows_plugin.vcxproj]
```

> Downgrading works in `Debug`, but not in `Release`.
> 
> `fatal error C1083: Cannot open include file: 'wil/com.h': No such file or directory [D:\Apps\exampleapp\build\windows\plugins\webview_windows\webview_windows_plugin.vcxproj]`

I've managed to make it work. Here's what I did.
The problem lies in the fact that the `wil` library (`Windows Implementation Library`) is missing from the plugin compilation, for some reason.
You cannot add it to the built plugin project, as it deleted and recompiled everytime.
As you may know, the plugin's `Additional Include Directories` also look inside the project's `windows/flutter/ephemeral` folder, which is deleted after every `flutter clean`.
My solution was to modify the `CMakeLists.txt` file to copy the library in the `ephemeral` folder at every build.

So,

1. Download `Windows Implementation Library` by nuget ([link](https://www.nuget.org/packages/Microsoft.Windows.ImplementationLibrary/))
2. Install it with Visual Studio, and go to the `%userprofile%/.nuget/packages/microsoft.windows.implementationlibrary/1.0.220914.1/include` folder.
3. Copy the `wil` folder inside the project's `windows/runner/resources/`
4. Edit the `windows/flutter/CMakeLists.txt` by inserting the following lines.
```cmake
# TODO: Move the rest of this into files in ephemeral. See
# https://github.com/flutter/flutter/issues/57146.
set(WRAPPER_ROOT "${EPHEMERAL_DIR}/cpp_client_wrapper")

# INSERT THIS LINE HERE
file(COPY "${PROJECT_DIR}/windows/runner/resources/wil" DESTINATION "${EPHEMERAL_DIR}/")

# === Flutter Library ===
set(FLUTTER_LIBRARY "${EPHEMERAL_DIR}/flutter_windows.dll")
```

Hope it helps until the `0.2.2` version will work with a future VS2022 release.

> While Running facing below error. Flutter verion: 3.7.5 Visual Studio: 2022
> 
> ```
> [ +685 ms] std.ixx.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) void (__cdecl* std::_Raise_handler)(class
> stdext::exception const &)" (__imp_?_Raise_handler@std@@3P6AXAEBVexception@stdext@@@ZEA::<!std>) referenced in function "public: void    
> __cdecl stdext::exception::_Raise(void)const " (?_Raise@exception@stdext@@QEBAXXZ::<!std>)
> [C:\Users\Administrator\Documents\codecommit\url_host\build\windows\plugins\webview_windows\webview_windows_plugin.vcxproj]
> ```

version 0.2.2, change file CMakeLists.txt, line 77, 
target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_23)
to
target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_20)
can solve this problem

> > While Running facing below error. Flutter verion: 3.7.5 Visual Studio: 2022
> > ```
> > [ +685 ms] std.ixx.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) void (__cdecl* std::_Raise_handler)(class
> > stdext::exception const &)" (__imp_?_Raise_handler@std@@3P6AXAEBVexception@stdext@@@ZEA::<!std>) referenced in function "public: void    
> > __cdecl stdext::exception::_Raise(void)const " (?_Raise@exception@stdext@@QEBAXXZ::<!std>)
> > [C:\Users\Administrator\Documents\codecommit\url_host\build\windows\plugins\webview_windows\webview_windows_plugin.vcxproj]
> > ```
> 
> version 0.2.2, change file CMakeLists.txt, line 77, target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_23) to target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_20) can solve this problem

This worked for me...
Thanks a lot

--- a/windows/CMakeLists.txt
+++ b/windows/CMakeLists.txt
@@ -72,12 +72,7 @@ else()
 endif()
 
 apply_standard_settings(${PLUGIN_NAME})
-if (MSVC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.29.30129 AND CMAKE_VERSION VERSION_GREATER 3.20.3)
-  # CMake 3.20.4+
-  target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_23) # ensure /std:c++latest for std::format
-else()
-  target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_20) # ensure /std:c++latest for std::format
-endif()
+target_compile_features(${PLUGIN_NAME} PUBLIC cxx_std_20) # For std::format support
 
 set_target_properties(${PLUGIN_NAME} PROPERTIES CXX_VISIBILITY_PRESET hidden)
 

